packages/forma-36-codemod/transforms/v4-badge.js

```diff
@@ -0,0 +1,56 @@
+const {
+  getComponentLocalName,
+  renameProperties,
+  changeProperties,
+  changeImportWithComponentName,
+  changeComponentName,
+} = require('../utils');
+const { getFormaImport, shouldSkipUpdateImport } = require('../utils/config');
+
+module.exports = function (file, api) {
+  const j = api.jscodeshift;
+
+  let source = file.source;
+
+  const componentName = getComponentLocalName(j, source, {
+    componentName: 'Tag',
+    importName: getFormaImport(),
+  });
+
+  if (!componentName) {
+    return source;
+  }
+
+  source = changeProperties(j, source, {
+    componentName,
+    fn(attributes) {
+      let modifiedAttributes = attributes;
+
+      modifiedAttributes = renameProperties(modifiedAttributes, {
+        renameMap: {
+          tagType: 'variant',
+        },
+      });
+
+      return modifiedAttributes;
+    },
+  });
+
+  source = changeComponentName(j, source, {
+    componentName,
+    outputComponentName: 'Badge',
+  });
+
+  // TODO: add and option to migrate <Tag entityStatusType="draft">Draft</Tag>; to <EntityStatusBadge entityStatus="draft" />;
```

Comment:
so, basically, we need to check if the component has `entityStatusType` property then we remove children and rename it to Badge?

Reply:
yeah, I guess that's it.

Reply:
I will give it a try.

Reply:
it could a generic `removeChildren` function or something like this. if you'll struggle implementing that, we can take a look together.